src/app/modules/analytics/__tests__/analytics.controller.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { DatabaseError } from 'dist/backend/app/modules/core/core.errors'
+import { errAsync, okAsync } from 'neverthrow'
+import expressHandler from 'tests/unit/backend/helpers/jest-express'
+
+import * as AnalyticsController from '../analytics.controller'
+import { AnalyticsFactory } from '../analytics.factory'
+import * as AnalyticsService from '../analytics.service'
+
+describe('analytics.controller', () => {
+  const MOCK_REQ = expressHandler.mockRequest()
+  afterEach(() => jest.clearAllMocks())
+
+  describe('handleGetUserCount', () => {
+    it('should return 200 with number of users on success', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+      const mockUserCount = 21
+      const getUserSpy = jest
+        .spyOn(AnalyticsService, 'getUserCount')
+        .mockReturnValueOnce(okAsync(mockUserCount))
+
+      // Act
+      await AnalyticsController.handleGetUserCount(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(getUserSpy).toHaveBeenCalledTimes(1)
+      expect(mockRes.json).toHaveBeenCalledWith(mockUserCount)
```

Comment:
i think you should tighten the test on `mockRes` as it's technically possible to call `res.status(400).json(...)`

Reply:
perhaps something like this?
```suggestion
      expect(mockRes.json).toHaveBeenCalledWith(mockUserCount)
      expect(mockRes.status).not.toHaveBeenCalled()
```

Reply:
Good point, will update all the missing asserts in this test file